source/_posts/angular-session-storage.md

```diff
@@ -0,0 +1,315 @@
+
+{% img "view-data.jpg" "View Data in angular session storage" "lazy" %}
+
+### Deleting data based on key
+
+For deleting the data saved in session storage based on the key, first, we will create a `dataRemove()` function in the `app.component.ts` file.
+
+```typescript
+dataRemove(){
+}
+```
+
+Inside the `dataRemove()` function add `removeItem` to remove specific data from the `sessionStorage` based on key.
+
+Syntax:
+
+```typescript
+localStorage.removeItem("key");
+```
+
+Thus, this is how `app.component.ts` will appear:
+
+```typescript
+dataRemove(){
+  sessionStorage.removeItem('name');
+}
+```
+
+After this, add a button in `app.component.html` file that'll help us remove the stored data.
+
+```html
+<button (click)="dataRemove()">Remove data from sessionStorage</button>
+```
+
+{% img "remove-data.jpg" "Remove data in angular session storage" "lazy" %}
+
+> Note: To clarify things, you can store multiple data in session storage and then use this method to remove specific data.
+
+For example, I've added a few additional data to our session storage, and I'll specifically remove the data with key value as Name.
+
+```typescript
+dataSave(){
+    sessionStorage.setItem('Name', 'Sanjeev');
+    sessionStorage.setItem('Country', 'India');
+    sessionStorage.setItem('Age', '22');
+  }
+dataRemove(){
+    sessionStorage.removeItem('Name');
+```
+
+### Delete all data from session storage
+
+For deleting all data from session storage first we'll create a `deleteAll()` function in the `app.component.ts` file.
+
+```typescript
+deleteAll(){
+}
+```
+
+Now in this function, we will use clear to delete all data from session storage.
+
+Syntax:
+
+```typescript
+deleteAll(){
+  sessionStorage.clear();
+}
+```
+
+Now, add a button to clear all data in the `app.component.html` file with a click event.
+
+```html
+<button (click)="deleteAll()" >Clear sessionStorage</button>
+```
+
+{% img "clear-data.jpg" "Clear all data in angular session storage" "lazy" %}
+
+In the end, our files will look like this:
+
+
+
+```typescript app.component.ts
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  title = 'new_project';
+
+  dataSave(){
+    sessionStorage.setItem('name', 'Sanjeev');
+  }
+
+  get(){
+    return sessionStorage.getItem('name');
+  }
+
+  dataRemove(){
+    sessionStorage.removeItem('name');
+  }
+
+  deleteAll(){
+    sessionStorage.clear();
+  }
+}
+```
+    
+```html app.component.html
+<h1>Angular sessionStorage</h1>
+<button (click)="dataSave()">Save data to sessionStorage</button>
+<p>{{ get() }}</p>
+<button (click)="dataRemove()">Remove data from sessionStorage</button>
+<button (click)="deleteAll()" >Clear sessionStorage</button>
+```
+
+## Angular Local Storage
+
+A `localstorage` object can be set, accessed, removed, or cleared with a single line of code, just like session storage.

```

Comment:
A `localstorage` should start with uppercase letter for `storage' word. Please fix it everywhere. There is no 'localstorage' object in Javascript.